packages/mdl-drawer/temporary/index.js

```diff
@@ -0,0 +1,138 @@
+
+    return root;
+  }
+
+  static attachTo(root) {
+    return new MDLTemporaryDrawer(root);
+  }
+
+  /* Remap touch events to pointer events, if the browser supports them. */
+  static remapEvent(eventName) {
+    if (this.usePointerEvents_ === undefined) {
+      this.usePointerEvents_ = Boolean(window.PointerEvent);
+    }
+
+    if (this.usePointerEvents_) {
+      switch (eventName) {
+        case 'touchstart':
+          return 'pointerdown';
+        case 'touchmove':
+          return 'pointermove';
+        case 'touchend':
+          return 'pointerup';
+        default:
+          return eventName;
+      }
+    }
+
+    return eventName;
+  }
+
+  static get transformPropertyName() {
+    if (this.transformPropertyName_ === undefined) {
+      const el = document.createElement('div');
+      this.transformPropertyName_ = ('transform' in el.style ? 'transform' : '-webkit-transform');
+    }
+    return this.transformPropertyName_;
+  }
+
+  static get supportsCSSCustomProperties() {
+    if (this.supportsCSSCustomProperties_ === undefined) {
+      this.supportsCSSCustomProperties_ = CSS.supports('(--color: red)');
+    }
+    return this.supportsCSSCustomProperties_;
+  }
+
+  get open() {
+    return this.foundation_.isOpen();
+  }
+
+  set open(value) {
+    if (value) {
+      this.foundation_.open();
+    } else {
+      this.foundation_.close();
+    }
+  }
+
+  /* Return the drawer element inside the component. */
+  get drawer() {
+    return this.root_.querySelector(MDLTemporaryDrawerFoundation.strings.DRAWER_SELECTOR);
+  }
+
+  getDefaultFoundation() {
+    const {FOCUSABLE_ELEMENTS, OPACITY_VAR_NAME, TAB_DATA, TAB_DATA_HANDLED} = MDLTemporaryDrawerFoundation.strings;
+
+    return new MDLTemporaryDrawerFoundation({
+      addClass: className => this.root_.classList.add(className),
+      removeClass: className => this.root_.classList.remove(className),
+      hasClass: className => this.root_.classList.contains(className),
+      hasNecessaryDOM: () => Boolean(this.drawer),
+      registerComponentInteractionHandler: (evt, handler) =>
+          this.root_.addEventListener(MDLTemporaryDrawer.remapEvent(evt), handler),
+      deregisterComponentInteractionHandler: (evt, handler) =>
+          this.root_.removeEventListener(MDLTemporaryDrawer.remapEvent(evt), handler),
+      registerDrawerInteractionHandler: (evt, handler) =>
+          this.drawer.addEventListener(MDLTemporaryDrawer.remapEvent(evt), handler),
+      deregisterDrawerInteractionHandler: (evt, handler) =>
+          this.drawer.removeEventListener(MDLTemporaryDrawer.remapEvent(evt), handler),
+      registerTransitionEndHandler: handler => this.drawer.addEventListener('transitionend', handler),
+      deregisterTransitionEndHandler: handler => this.drawer.removeEventListener('transitionend', handler),
+      getDrawerWidth: () => this.drawer.offsetWidth,
+      updateDrawerPosition: value => this.drawer.style.setProperty(
```

Comment:
since this method is called `updateDrawerPosition` and `position` is a property in CSS with a different coordinate system than a translation transform, it may not be immediately be apparent to developers that one is supposed to alter a translation value here. Perhaps we should call it something like `setTranslateX` instead?